### ISSUE TITLE
Feature request: Restrict data to download by date

### ISSUE THREAD

**Issue by jcvdav** [eliocamp/rcmip6#7]
Hola @eliocamp - Muchas gracias por desarrollar este genial recurso.

I'll be switching over to English in case others are interested in this feature. I often find myself needing CMIP6 data just a few years into the future (~typically 2050 for my work), so there is no need to download all files from a given model, especially for the ones that go out a couple centuries.

Would it be possible to include a `max_year` argument in [`cmip_download()`](https://github.com/eliocamp/rcmip6/blob/main/R/cmip-download.R#L12)? In practice, the argument would have to then be passed to the helper at  [`cmip_download_one()`](https://github.com/eliocamp/rcmip6/blob/main/R/cmip-download.R#L52), where one can use the information in [`url`](https://github.com/eliocamp/rcmip6/blob/main/R/cmip-download.R#L62) to extract the first year contained in a file's data, evaluate it agains't `max_year`, and then decide whether to proceed or not. For example, starting at [`L73`](https://github.com/eliocamp/rcmip6/blob/main/R/cmip-download.R#L73), the code could be modified to have:

```
.
.
.

files <-  vapply(info, function(i) {
    url <- strsplit(i$url[[1]], "\\|")[[1]][1]
    date <- as.numeric(substr(strsplit(strsplit(url, "gn\\_")[[1]][2], "-")[[1]][1], 1, 4))
    
    if (date > max_year) {
      return(NA_character_)
    }
.
.
.
}
```

If you think t his would be valuable, and the right implementation, I'm happy to submit a pull request for your evaluation.

JC

**Comment by eliocamp** [comment]
You should filter the results as you would any data.frame (using `subset()`, or `dplyr::filter()` or what have you) and then use `cmip_download()` to download the things. 

Perhaps the issue is that the results don't have a proper "date" column for you to filter on. I though they have a "start_date" and "end_date" columns. Can't you filter on those?

**Comment by jcvdav** [comment]
You are correct in that the results of `cmip_download()` contain an `end_date` column, but this is the end date of the entire simulation. For example, the first row of the output shown in the [README example](https://github.com/eliocamp/rcmip6#rcmip6) shows that the historical run includes data between `date_start = 1850-01-16T12:00:00Z`  and `date_end = 2014-12-16T12:00:00Z`. Calling `cmip_download()` would lead to the download of all `*.nc` files between this period, which may be more than a user requires.

As it stands, there is no way for a user to restrict the download to get files corresponding to a period of interest (say, 1850 - 1900 only).

**Comment by eliocamp** [comment]
Oh, ok, I think I understand. This would apply to multi-file models, right?

I guess it makes sense. I think it would need to be a range of dates and the package would make sure to download the necessary files so that the ranges of dates are included. This would allow you to filter for max year and min year.

**Comment by jcvdav** [comment]
Yes, this is only relevant to multi-file models. For example, I all the models I am using (GFDL-ESM4, ACCESS-CM2, ACCESS-ESM1-5, BCC-CSM2-MR, CanESM5, EC-Earth3, MIROC6, MRI-ESM2-0, NorESM2-LM, NorESM2-MM) have either one file per decade or one file per year (I think).

And you are correct. A range of dates (`year_start` and `year_end`) would perhaps be more appropriate.

Let me know if you think this would be valuable or if you'd like me to submit a PR with a proposed approach.

**Comment by eliocamp** [comment]
Closed by #8
